File: crates/corelink-wasm/examples/quickstart_audit.ts
// quickstart_audit — Stream audit events via GET /v1/admin/audit-events.
//
// Customer concept: every privileged op is Merkle-chained into the
// audit log (INV-AUDIT-APPEND-ONLY). Response includes chain_head_hash
// for client-side verification (CTRL-AUDIT-002).
//
// Run: CORELINK_PAT=$ADMIN_PAT tsx examples/quickstart_audit.ts

const api = process.env.CORELINK_API_URL ?? "https://sandbox.corelink.dev";
const pat = process.env.CORELINK_PAT;
if (!pat) {
  console.error("error: CORELINK_PAT env var is required");
  process.exit(2);
}

const url = new URL(`${api}/v1/admin/audit-events`);
url.searchParams.set("limit", "50");

try {
  const resp = await fetch(url, {
    method: "GET",
    headers: { Authorization: `Bearer ${pat}`, Accept: "application/json" },
  });
  if (!resp.ok) {
    const body = await resp.text();
    console.error(`server returned ${resp.status}: ${body}`);
    process.exit(1);
  }
  const payload = (await resp.json()) as { events?: unknown[]; chain_head_hash?: string };
  const count = Array.isArray(payload.events) ? payload.events.length : 0;
  const head = payload.chain_head_hash ?? "<none>";
  console.log(`status=${resp.status} events=${count} chain_head=${head}`);
} catch (err) {
  console.error(`network_error: ${err instanceof Error ? err.message : String(err)}`);
  process.exit(1);
}

File: tools/cli/examples/quickstart_audit.rs
//! `quickstart_audit` — Stream audit events via `GET /v1/admin/audit-events`.
//!
//! Customer concept: every privileged operation in CoreLink is
//! Merkle-chained into the audit log (INV-AUDIT-APPEND-ONLY). This
//! endpoint paginates the chain in commit order; the response includes
//! `chain_head_hash` for client-side verification (CTRL-AUDIT-002).
//!
//! Run:
//!
//! ```text
//! CORELINK_PAT=$ADMIN_PAT \
//!   cargo run --example quickstart_audit -p corelink-cli
//! ```

#![forbid(unsafe_code)]
#![allow(
    clippy::print_stdout,
    reason = "examples emit human-readable narration to stdout"
)]

use anyhow::{Context, Result};
use std::env;

#[tokio::main]
async fn main() -> Result<()> {
    let api = env::var("CORELINK_API_URL")
        .unwrap_or_else(|_| "https://sandbox.corelink.humangr.com".to_string());
    let pat = env::var("CORELINK_PAT").context("CORELINK_PAT is required")?;

    let resp = reqwest::Client::new()
        .get(format!("{api}/v1/admin/audit-events"))
        .bearer_auth(&pat)
        .query(&[("limit", "50")])
        .send()
        .await
        .context("GET /v1/admin/audit-events")?;

    let status = resp.status();
    if !status.is_success() {
        let body = resp.text().await.unwrap_or_default();
        anyhow::bail!("server returned {status}: {body}");
    }
    let body: serde_json::Value = resp.json().await.context("decode body")?;
    let events = body
        .get("events")
        .and_then(|v| v.as_array())
        .map_or(0, Vec::len);
    let head = body
        .get("chain_head_hash")
        .and_then(|v| v.as_str())
        .unwrap_or("<none>");
    println!("status={status} events={events} chain_head={head}");
    Ok(())
}
